bigtable-hbase-mirroring-client-2.x-parent/bigtable-hbase-mirroring-client-2.x/src/test/java/com/google/cloud/bigtable/mirroring/hbase2_x/TestMirroringAsyncTable.java

```diff
@@ -0,0 +1,831 @@
+    RowMutations mutations = new RowMutations("r1".getBytes());
+    when(primaryBuilder.thenMutate(mutations)).thenReturn(CompletableFuture.completedFuture(true));
+    mirroringTable.checkAndMutate("r1".getBytes(), "f1".getBytes()).thenMutate(mutations).get();
+    verify(secondaryTable, times(1)).mutateRow(mutations);
+  }
+
+  @Test
+  public void testDelete() throws InterruptedException, ExecutionException {
+    Delete delete = new Delete("r1".getBytes());
+    CompletableFuture<Void> primaryFuture = new CompletableFuture<>();
+    CompletableFuture<Void> secondaryFuture = new CompletableFuture<>();
+    when(primaryTable.delete(delete)).thenReturn(primaryFuture);
+    when(secondaryTable.delete(delete)).thenReturn(secondaryFuture);
+
+    CompletableFuture<Void> resultFuture = mirroringTable.delete(delete);
+    primaryFuture.complete(null);
+    secondaryFuture.complete(null);
+    resultFuture.get();
+    verify(secondaryTable, times(1)).delete(delete);
+  }
+
+  @Test
+  public void testMutateRow() throws ExecutionException, InterruptedException {
+    RowMutations mutations = new RowMutations("r1".getBytes());
+    CompletableFuture<Void> primaryFuture = new CompletableFuture<>();
+    CompletableFuture<Void> secondaryFuture = new CompletableFuture<>();
+    when(primaryTable.mutateRow(mutations)).thenReturn(primaryFuture);
+    when(secondaryTable.mutateRow(mutations)).thenReturn(secondaryFuture);
+
+    CompletableFuture<Void> resultFuture = mirroringTable.mutateRow(mutations);
+    primaryFuture.complete(null);
+    secondaryFuture.complete(null);
+    resultFuture.get();
+    verify(secondaryTable, times(1)).mutateRow(mutations);
+  }
+
+  private void assertPutsAreEqual(Put expectedPut, Put value) {
+    TestHelpers.assertPutsAreEqual(
+        expectedPut, value, (a, b) -> CellComparator.getInstance().compare(a, b));
+  }
+
+  @Test
+  public void testIncrement() throws ExecutionException, InterruptedException {
+    Increment increment = new Increment("r1".getBytes());
+    Result incrementResult =
+        Result.create(
+            new Cell[] {
+              CellBuilderFactory.create(CellBuilderType.SHALLOW_COPY)
+                  .setRow("r1".getBytes())
+                  .setFamily("f1".getBytes())
+                  .setQualifier("q1".getBytes())
+                  .setTimestamp(12)
+                  .setType(Cell.Type.Put)
+                  .setValue(Longs.toByteArray(142))
+                  .build()
+            });
+    Put expectedPut = OperationUtils.makePutFromResult(incrementResult);
+
+    // increment() and append() modify the reference counter twice to make logic less brittle
+    when(primaryTable.increment(any(Increment.class)))
+        .thenReturn(CompletableFuture.completedFuture(incrementResult));
+    verify(referenceCounter, never()).decrementReferenceCount();
+    verify(referenceCounter, never()).incrementReferenceCount();
+    mirroringTable.increment(increment).get();
```

Comment:
Please add comments on the flow of this test. It is not clearn to me what is the increment doing vs what is incrementColumnValue doing (both on the mirroring table)